### PR TITLE
[Doc] add explanations and references for all clients

### PR DESCRIPTION
**Pull request by Anonymitaet**
Preview looks good:
image

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@BewareMyPower @codelipenghui could you please help review it from the technical perspective? Thanks 

@DaveDuggins @momo-jun could you please help review it from the technical writing perspective? Thanks

**Review by BewareMyPower — APPROVED** [review]

**Review by Huanli-Meng — APPROVED** [review]

File: site2/docs/getting-started-clients.md [hunk 1]
```diff
@@ -6,13 +6,16 @@ sidebar_label: Overview
...
+Python | - [User doc](client-libraries-python.md) <br /><br />- [API doc](https://pulsar.apache.org/api/python/)|Python client is released along with Pulsar, so Python client release note is a part of [Pulsar release note](https://pulsar.apache.org/release-notes/).|[Python client code repo](https://github.com/apache/pulsar/tree/master/pulsar-client-cpp/python) is hosted in the Pulsar main repo on GitHub. 
+WebSocket| [User doc](client-libraries-websocket.md) | WebSocket client is released along with Pulsar, so WebSocket client release note is a part of [Pulsar release note](https://pulsar.apache.org/release-notes/).|[WebSocket client code repo](https://github.com/apache/pulsar/tree/master/pulsar-websocket) is hosted in the Pulsar main repo on GitHub.
+Go client|[User doc](client-libraries-go.md)|Go client is not released along with Pulsar. The release note is available at [here](docs.google.com/document/d/1xiptFfsOvD1kHqGv6FyeMTD41R7Dffbhcz_1kCT_Rk0/edit#). |[Go client code repo](https://github.com/apache/pulsar-client-go) is hosted outside of Pulsar main repo.
```
Comment (dave2wave) [inline site2/docs/getting-started-clients.md:15]: This link is to some other google doc and is in edit mode.

1. Release notes should be in a repository.
2. Release notes should not be in an immediately editable file format.

**Review by dave2wave — CHANGES_REQUESTED** [review]

File: site2/docs/getting-started-clients.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/docs/getting-started-clients.md]: @dave2wave Thanks for your suggestion! I've updated, PTAL

**Review by dave2wave — APPROVED** [review]
LGTM

**Review by eolivelli — APPROVED** [review]

**Pull request merged**